### PR TITLE
Include `symbol-annotation` in core BOM

### PR DESCRIPTION
**Pull request by basil**
See [jenkinsci/jenkins#5293 (comment)](https://github.com/jenkinsci/jenkins/pull/5293#issuecomment-780542952). Now that `symbol-annotation` has been [split into its own repository](https://github.com/jenkinsci/lib-symbol-annotation), encourage plugins to use the dependency from Jenkins core rather than the `structs` plugin by adding `symbol-annotation` to the core (not plugin!) BOM. The corresponding change to stop bundling this with `structs` is jenkinsci/structs-plugin#112. Note that I am not upgrading the version of this library yet. I want to wait until jenkinsci/structs-plugin#112 is released and more plugins adopt the new release of structs before I do that in order to avoid unnecessary migration pain.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [X] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)).

### CODE REVIEW

File: test/pom.xml
Comment (jglick) [inline test/pom.xml:138]: This must be in draft so long as that is unreleased, since the incremental build could be collected at some point and we would not want this dep in `master`. See https://github.com/jenkinsci/pom/pull/172.

Comment (timja) [reply test/pom.xml:138]: this could be separated out to another PR I guess.

File: test/pom.xml
Comment (jglick) [inline test/pom.xml:141]: Or just wait for @car-roll to release https://github.com/jenkinsci/structs-plugin/pull/112?

**Review by jglick — APPROVED** [review]

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
